### ISSUE TITLE
Does Not Work With Node 7.9.0

### ISSUE THREAD

**Issue by MRayermannMSFT** [capriza/syswide-cas#1]
We recently upgrade to Node 7.9.0 and have started receiving SSL certificate related errors in environments where we were using this package to extend the Node certificate authority. I believe that the reason for these errors is an incompatibility between this package and Node 7.9.0.

Unfortunately I am unable to share my full repro case as the environment is private, but in general I tested it by doing the following:

1. Create a Node script which uses syswide-cas to add certificates and then make a request to a server with self signed certificates:
``` javascript
const syswidecas = require('syswide-cas');
const request = require('request');

syswidecas.addCAs("path to directory with certificates");

var endpoint = "redacted";
var accessToken = "redacted";

request.get(
    endpoint,
    {
        headers: {
            "Content-Type": "application/json",
            "Authorization": "Bearer " + accessToken
        },
        json: true
    },
    (err, res, body) => {
        if (err) {
            console.log("err: " + err);
        } else {
            console.log(JSON.stringify(body))
        }
    }
)
;
```
In this case, I received a `unable to get local issuer certificate` error.

2. Create a Node script which uses the `ca` option for `request.get` to override Node's certificate authority and then make a request to a server with self signed certificates:
``` javascript
const fs = require('fs');
const request = require('request');

var endpoint = "redacted";
var accessToken = "redacted";
var pemCerts = fs.readFileSync("path to PEM file", {encoding: 'ascii'}).trim();

request.get(
    endpoint,
    {
        headers: {
            "Content-Type": "application/json",
            "Authorization": "Bearer " + accessToken
        },
        ca: pemCerts,
        json: true
    },
    (err, res, body) => {
        if (err) {
            console.log("err: " + err);
        } else {
            console.log(JSON.stringify(body))
        }
    }
);
```
In this case, the request was successful.

We are aware of the recent addition of the `NODE_EXTRA_CA_CERTS` environment variable to Node, but we would prefer to continue using this package if possible.

**Comment by MRayermannMSFT** [comment]
It seems that the method being used by this package to add certs to the root CA was fixed in [this change ](https://github.com/nodejs/node/commit/bcdbf22f0de0e6a08b1743d274c215934bd7d71a) to Node.

**Comment by fujifish** [comment]
I seems the current way of doing it is no longer possible since node is now creating a new root cert store with default certs every time a cert is manually added to a secure context. 
I'm still trying to figure out how to go about this.

**Comment by fujifish** [comment]
I'v released version 4.0.0 that utilizes a different method for adding the custom root CA.
I've tested with the following code on versions 0.10, 6, 7.1, 7.2, 7.9 and 8.
Let me know if it works out.

```javascript
const selfsigned = "https://self-signed.badssl.com/";
const mozillaModern = "https://mozilla-modern.badssl.com/";

const syswidecas = require("syswide-cas");
syswidecas.addCAs("./wildcardSelfSigned.pem");

const request = require("request");

request.get(selfsigned, function(err, res, body) {
  if (err) {
    console.log(selfsigned + " error: " + err);
  } else {
    console.log(selfsigned + " success. body length: " + body.length);
  }

  request.get(mozillaModern, function(err, res, body) {
    if (err) {
      console.log(mozillaModern + " error: " + err);
    } else {
      console.log(mozillaModern + " success. body length: " + body.length);
    }
  });
});
```

The pem file for self-seigned.badssl.com:
```
-----BEGIN CERTIFICATE-----
MIIDeTCCAmGgAwIBAgIJAIb7Tcjl3Q8YMA0GCSqGSIb3DQEBCwUAMGIxCzAJBgNV
BAYTAlVTMRMwEQYDVQQIDApDYWxpZm9ybmlhMRYwFAYDVQQHDA1TYW4gRnJhbmNp
c2NvMQ8wDQYDVQQKDAZCYWRTU0wxFTATBgNVBAMMDCouYmFkc3NsLmNvbTAeFw0x
NjA4MDgyMTE3MDVaFw0xODA4MDgyMTE3MDVaMGIxCzAJBgNVBAYTAlVTMRMwEQYD
VQQIDApDYWxpZm9ybmlhMRYwFAYDVQQHDA1TYW4gRnJhbmNpc2NvMQ8wDQYDVQQK
DAZCYWRTU0wxFTATBgNVBAMMDCouYmFkc3NsLmNvbTCCASIwDQYJKoZIhvcNAQEB
BQADggEPADCCAQoCggEBAMIE7PiM7gTCs9hQ1XBYzJMY61yoaEmwIrX5lZ6xKyx2
PmzAS2BMTOqytMAPgLaw+XLJhgL5XEFdEyt/ccRLvOmULlA3pmccYYz2QULFRtMW
hyefdOsKnRFSJiFzbIRMeVXk0WvoBj1IFVKtsyjbqv9u/2CVSndrOfEk0TG23U3A
xPxTuW1CrbV8/q71FdIzSOciccfCFHpsKOo3St/qbLVytH5aohbcabFXRNsKEqve
ww9HdFxBIuGa+RuT5q0iBikusbpJHAwnnqP7i/dAcgCskgjZjFeEU4EFy+b+a1SY
QCeFxxC7c3DvaRhBB0VVfPlkPz0sw6l865MaTIbRyoUCAwEAAaMyMDAwCQYDVR0T
BAIwADAjBgNVHREEHDAaggwqLmJhZHNzbC5jb22CCmJhZHNzbC5jb20wDQYJKoZI
hvcNAQELBQADggEBALW4pad52T7VNw2nFMjPH98ZJNAQQgWyr3H2KlZN6IFGsonO
nCC/Do8BPx6BnP3PFwovWMat1VvnRRoC8lw/30eEazWqBRGZWPz6LHTE3DNBJdc8
xz6mh8q9RJX/PAj+YYGNElTu6qj49YT0BEhMF4U+dTQ0G8y3x4WNfiu9pGqyrp8d
AzeidMfQ/pU01PpoPTDLvRDNkmMsABNE1fXBfJxDDGwfq1xY1j23Fm6BolwZC2y7
n19h+vMYVWbGoovrf2/ibTvtcTyfDop7gl5Yy3OncZxokFj21rUZpLgx9ea4a9z3
FzEz5ufynq03RhHTE1eu+gDzMEF0GNhGGsKqeA4=
-----END CERTIFICATE-----
```

**Comment by MRayermannMSFT** [comment]
Hey @fujifish, the new release works great! Thank you so much for such quick turnaround on this issue. :)
